contracts/protocols/mainnet/erc4626/main.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+import "./interfaces.sol";
+
+contract Resolver {
+    struct VaultData {
+        bool isToken;
+        string name;
+        string symbol;
+        uint256 decimals;
+        address asset;
+        uint256 totalAssets;
+    }
+
+    function getVaultDetails(address vaultAddress) public view returns (VaultData memory) {
+        VaultData memory _vaultData;
+        VaultInterface vaultToken = VaultInterface(vaultAddress);
+        bool isToken = true;
+
+        try vaultToken.symbol() {} catch {
+            isToken = false;
+        }
+
+        try vaultToken.name() {} catch {
+            isToken = false;
+        }
+
+        try vaultToken.decimals() {} catch {
+            isToken = false;
+        }
+
+        _vaultData = VaultData(
+            isToken,
+            vaultToken.name(),
+            vaultToken.symbol(),
+            vaultToken.decimals(),
+            vaultToken.asset(),
+            vaultToken.totalAssets()
```

Comment:
Hey! So as resolvers are used for read calls. You can return all the data. For example, convert to assets or convert to shares as well. As this is also some data we will show users on UI. Just try to return as much things as possible as the gas is not an issue here.

Reply:
For convert to assets & convert to shares. You can use 1 token for example 10**decimals.

Reply:
Also, the data of `convert to assets` & `convert to shares` also helps on the UI level when user wants to deposit/mint & withdraw/redeem to set the minimum or maximum limit after particular action.